elastic/shared/runners/ilm.py

```diff
@@ -35,7 +35,12 @@ async def create_ilm(es, params):
             name = os.path.splitext(os.path.basename(p))[0]
             with open(p, "r") as f:
                 policy = {"policy": json.load(f)["policy"]}  # only `policy` is allowed
-            await es.ilm.put_lifecycle(name, body=policy)
+            try:
+                await es.ilm.put_lifecycle(name=name, policy=policy)
+            # method signature changed in es client >8.x
+            except TypeError:
+                await es.ilm.put_lifecycle(name, body=policy)
+
```

Comment:
Something I don't like about those try/except (including the one I suggested below!) is that we don't have a clear end-of-life policy for tracks. For how long do we have to support using Rally 2.7.0? On what branches?

One reason I'm saying this is that we can get avoid try/excepts completely here:

 * This specific change could be handled in Rally in the same way that we changed the signature of perform_request in https://github.com/elastic/rally/pull/1512
 * We could just catch `Exception` in remote_cluster.py. With exception chaining even if there's a TypeError we'll see it so making sure we only catch exceptions from the Python client becomes less important.

Reply:
Agreed - I added the pre-requisite [BWC logic to Rally as part of the 8.x upgrade](https://github.com/elastic/rally/pull/1669/commits/b62d79362ee713f587ffee618234ac2475d10806) and I've reverted the track/ILM runner specific BWC logic in https://github.com/elastic/rally-tracks/pull/380/commits/52d036a1fdad81a3172a342640a03a6b23852e60